### PR TITLE
add MultiRegionTimeseriesDataset

### PR DESCRIPTION
**Pull request by TomGoBravo**
* Adds frozen final dataclass `MultiRegionTimeseriesDataset` which replaces `TimeseriesDataset` and reduces places where `data` is directly accessed, making it easier to change how regions are represented within the dataframe.
* Removes `combined_datasets.get_timeseries_for_fips`
* Removes `infer_t0.py`

This PR is part of https://trello.com/c/nYgDEl3M/385-ability-to-query-raw-data-by-msa

### CODE REVIEW

File: libs/pipeline.py [hunk 1]
```diff
@@ -10,6 +10,13 @@
 import us
 
 
+def fips_to_location_id(fips: str) -> str:
+    """Converts a FIPS code to a locationID"""
+    if len(fips) == 2:
+        iso1: us  # iso2:us-ca#fips:06087
```
Comment (ghop02) [inline libs/pipeline.py:16]: This feels incomplete?

File: libs/datasets/timeseries.py [hunk 2]
```diff
@@ -292,6 +282,62 @@ def load_csv(cls, path_or_buf: Union[pathlib.Path, TextIO]):
         return cls(df)
 
 
+@final
+@dataclass(frozen=True)
+class MultiRegionTimeseriesDataset:
```
Comment (ghop02) [inline libs/datasets/timeseries.py:287]: The timeseries.py file is getting a bit crowded, thoughts on when we should separate the existing classes into multiple files?

**Review by ghop02 — APPROVED** [review]
seems fine to me right now!

File: libs/pipeline.py (hunk 1 above)
Comment (TomGoBravo) [reply libs/pipeline.py]: oops, that's from another branch. fixed with 2a8fcdd

File: libs/datasets/timeseries.py (hunk 2 above)
Comment (TomGoBravo) [reply libs/datasets/timeseries.py:287]: I'd like to replace the last of TimeseriesDataset with the more static and specific OneRegion/MultiRegionTimeseriesDataset, shrinking this file back down a lot.

**Pull request merged**